packages/block-library/src/gallery/gallery-image.js

```diff
@@ -184,12 +184,15 @@ class GalleryImage extends Component {
 			'is-selected': isSelected,
 			'is-transient': isBlobURL( url ),
 		} );
-
+		// Safari: give the figure a tab index to make it focusable,
+		// so that the blur event is canceled when a child inside it is focused (see this.onFocus)
+		// see: https://github.com/Automattic/wp-calypso/issues/40531#issuecomment-617384307
 		return (
 			<figure
 				className={ className }
 				onBlur={ this.onBlur }
 				onFocus={ this.onFocus }
+				tabIndex="0"
```

Comment:
tabindex can have unexpected impact on keyboard navigation... Did you consider whether we could avoid it and add a check on the onBlur call and only run the behavior if we really leave the block? 

cc @nosolosw as this seems very related to what we've been discussing on a previous PR.

Reply:
True for `tabIndex`, I don't think it's true for `tabIndex=0`

> tabindex="0" means that the element should be focusable in sequential keyboard navigation, but its order is defined by the document's source order.

From: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex

Reply:
I'm sorry I can't test and debug this on Safari. I'm happy to share my thinking:

It looks like the issue here is [a bug in Safari](https://bugs.webkit.org/show_bug.cgi?id=22261) by which buttons don't trigger the `onFocus` event when clicked, but they do when they gain focus via tabbing. If that's correct, would this be fixed if we fire the `this.onFocus` event on the button `onClick` when the browser is Safari?

Reply:
The reason why this would be happening is a race-condition created by the [change introduced here](https://github.com/WordPress/gutenberg/pull/14930#discussion_r348383465): the image will be deselected if no focus event is fired by the figure children that gains the focus (hence why it'd fail in Safari), so any other action (move, remove) won't take effect. Can anyone bisect if that's the point where it started happening?

Reply:
I guess another thing that is happening is that any styles that are bound the `:focus` pseudo-class for buttons won't be applied either ([example](https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/gallery/editor.scss#L87)).